test_guide.md

```diff
@@ -0,0 +1,139 @@
+# How to write a node test
+
+## What is a test?
+
+A test must be a node script that executes a specific functionality provided by node and
+checks that it behaves as expected. It should return 0 on success, otherwise it will
+fail. A test will fail if:
+
+- It exits by calling `process.exit(code)` where `code != 0`
+- It exits due to an uncaught exception.
+- It never exits. In this case, the test runner will catch the error because it sets a maximum time limit.
+
+They can be added for multiple reasons:
+
+- when adding new functionality.
```

Comment:
The previous list started all the items with capital letters. This one is all lowercase. I think we should be consistent.